### DISCUSSION TITLE
Get api time from url

### DISCUSSION THREAD

**Discussion by priisdk** [micropython/micropython#9804] [General]
I use the following code snippet to get api time:

def getapitime():
    response=urequests.request('GET', "http://worldtimeapi.org/api/timezone/Europe/Copenhagen")
    parsed = response.json()
    return str(parsed["datetime"])

Sometimes it works but most often it does not (sometimes responding with "OSError: [Errno 104] ECONNRESET", sometimes responding with "IndexError: list index out of range"). How can I make it 100 % stable?

Poul Riis

**Comment by scruss** [comment]
100% stable is a long shot when network connections can't be, but this won't run out of memory on you:

```
def getapitime():
    try:
        response=urequests.request('GET', "http://worldtimeapi.org/api/timezone/Europe/Copenhagen")
        parsed = response.json()
        response.close()
    except:
        return "0000-00-00T00:00:00.0+01:00"
    return str(parsed["datetime"])
```
